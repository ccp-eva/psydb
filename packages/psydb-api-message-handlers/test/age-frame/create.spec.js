'use strict';
var { ejson, omit } = require('@mpieva/psydb-core-utils');
var { expect } = require('@mpieva/psydb-api-mocha-test-tools/chai');
var { ObjectId } = require('@cdxoo/mongo-test-helpers');
var jsonify = (that) => (
    JSON.parse(JSON.stringify(that))
);

var RootHandler = require('../../src/');

var omitNonsense = ({ from }) => omit({
    from, paths: [ '_id', '_rohrpostMetadata' ]
});

describe('ageFrame/create', function () {
    var db, sendMessage;
    beforeEach(async function () {
        await this.restore('2024-03-28__0556_humankind');
        
        db = this.getDbHandle();
        ([ sendMessage ] = this.createMessenger({
            RootHandler,
            ...(await this.createFakeLogin({ email: 'root@example.com' }))
        }));
    });

    it('does the thing', async function () {
        var studyId = ObjectId("6600f821a5f8769c528681df");
        var subjectSelectorId = ObjectId("6604cdcd8df35a1505347aac");

        var arabicId = ObjectId('6600f5e0a5f8769c52868145');
        var englishId = ObjectId('6600f5e0a5f8769c52868142');

        var koaContext = await sendMessage({
            type: 'ageFrame/create',
            timezone: 'Europe/Berlin',
            payload: jsonify({
                studyId,
                subjectSelectorId,
                subjectTypeKey: 'humankindChild',
                props: {
                    interval: {
                        start: { years: 0, months: 0, days: 0 },
                        end: { years: 1, months: 1, days: 1 },
                    },
                    conditions: [
                        { pointer: '/gdpr/state/custom/gender', values: [
                            'male', 'female'
                        ]},
                        { 
                            pointer: '/scientific/state/custom/nativeLanguageId',
                            values: [ arabicId, englishId ]
                        }
                    ]
                }
            })
        });

        var { body } = koaContext.response;
        var [ ageFrameUpdate ] = body.data;

        console.log(ageFrameUpdate);

        var record = omitNonsense({
            from: await this.getRecord('ageFrame', {
                _id: ageFrameUpdate.channelId
            })
        });

        console.dir(ejson(record), { depth: null });
        expect(ejson(record)).toMatchSnapshot();
    })
})
