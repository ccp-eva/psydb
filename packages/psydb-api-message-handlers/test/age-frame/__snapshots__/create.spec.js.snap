// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ageFrame/create does the thing 1`] = `
Object {
  "isDummy": false,
  "sequenceNumber": "14",
  "state": Object {
    "conditions": Array [
      Object {
        "pointer": "/gdpr/state/custom/gender",
        "values": Array [
          "male",
          "female",
        ],
      },
      Object {
        "pointer": "/scientific/state/custom/nativeLanguageId",
        "values": Array [
          Object {
            "$oid": "6600f5e0a5f8769c52868145",
          },
          Object {
            "$oid": "6600f5e0a5f8769c52868142",
          },
        ],
      },
    ],
    "interval": Object {
      "end": Object {
        "days": 1,
        "months": 1,
        "years": 1,
      },
      "start": Object {
        "days": 0,
        "months": 0,
        "years": 0,
      },
    },
  },
  "studyId": Object {
    "$oid": "6600f821a5f8769c528681df",
  },
  "subjectSelectorId": Object {
    "$oid": "6604cdcd8df35a1505347aac",
  },
  "subjectTypeKey": "humankindChild",
}
`;
